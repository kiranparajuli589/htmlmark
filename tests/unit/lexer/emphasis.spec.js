import { Lexer } from "../../../lib/lexer/index.js"


describe("emphasis", () => {
	describe("bold", () => {
		it("should match the consecutive occurrences", () => {
			const lines = [
				"**first bold** **second bold** **third bold**"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
		it("should be deep tokenized", () => {
			const lines = [
				"**bold with `code` and *italics* and ~~strikes~~ and [link](https://kiranparajuli.com.np)**"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
	})
	describe("italics", () => {
		it("should match the consecutive occurrences", () => {
			const lines = [
				"*first italics* *second italics* *third italics*"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
		it("should be deep tokenized", () => {
			const lines = [
				"*italics with `code` and ~~strikes~~ and [link](https://kiranparajuli.com.np)*"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
	})
	describe("strikes", () => {
		it("should match the consecutive occurrences", () => {
			const lines = [
				"~~first strikes~~ ~~second strikes~~ ~~third strikes~~"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
		it("should be deep tokenized", () => {
			const lines = [
				"~~strikes with ~ gem and `code` and **bold** and *italics* and [link](https://kiranparajuli.com.np)~~"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
	})
	describe("code", () => {
		it("should match the consecutive occurrences", () => {
			const lines = [
				"`first code` `second code` `third code`"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
		it("should not be deep tokenized", () => {
			const lines = [
				"`code with ~ gem and **bold** and *italics* and [link](https://kiranparajuli.com.np)`"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
	})
	describe("link", () => {
		it("should match the consecutive occurrences", () => {
			const lines = [
				"[link](https://kiranparajuli.com.np) [link](https://kiranparajuli.com.np)"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
		it("should deep tokenized the link title", () => {
			const lines = [
				"[*italic* with **bold** and `code`](https://kiranparajuli.com.np)"
			]
			const lexer = new Lexer(lines)
			const lexerData = lexer.run()
			expect(lexerData).toMatchSnapshot()
		})
	})
})
