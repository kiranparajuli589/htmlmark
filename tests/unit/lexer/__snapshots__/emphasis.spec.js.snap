// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emphasis bold should be deep tokenized 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "**bold with \`code\` and *italics* and ~~strikes~~ and [link](https://kiranparajuli.com.np)**",
    "tokens": Array [
      Object {
        "raw": "**bold with \`code\` and *italics* and ~~strikes~~ and [link](https://kiranparajuli.com.np)**",
        "tokens": Array [
          Object {
            "raw": "bold with ",
            "type": "text",
            "value": "bold with ",
          },
          Object {
            "raw": "\`code\`",
            "type": "code",
            "value": "code",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "*italics*",
            "tokens": Array [
              Object {
                "raw": "italics",
                "type": "text",
                "value": "italics",
              },
            ],
            "type": "italic",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "~~strikes~~",
            "tokens": Array [
              Object {
                "raw": "strikes",
                "type": "text",
                "value": "strikes",
              },
            ],
            "type": "strike-through",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "[link](https://kiranparajuli.com.np)",
            "tokens": Object {
              "href": "https://kiranparajuli.com.np",
              "title": Object {
                "raw": "link",
                "tokens": Array [
                  Object {
                    "raw": "link",
                    "type": "text",
                    "value": "link",
                  },
                ],
              },
              "tooltip": undefined,
            },
            "type": "link",
          },
        ],
        "type": "bold",
      },
    ],
    "type": "paragraph",
    "value": "**bold with \`code\` and *italics* and ~~strikes~~ and [link](https://kiranparajuli.com.np)**",
  },
]
`;

exports[`emphasis bold should match the consecutive occurrences 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "**first bold** **second bold** **third bold**",
    "tokens": Array [
      Object {
        "raw": "**first bold**",
        "tokens": Array [
          Object {
            "raw": "first bold",
            "type": "text",
            "value": "first bold",
          },
        ],
        "type": "bold",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "**second bold**",
        "tokens": Array [
          Object {
            "raw": "second bold",
            "type": "text",
            "value": "second bold",
          },
        ],
        "type": "bold",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "**third bold**",
        "tokens": Array [
          Object {
            "raw": "third bold",
            "type": "text",
            "value": "third bold",
          },
        ],
        "type": "bold",
      },
    ],
    "type": "paragraph",
    "value": "**first bold** **second bold** **third bold**",
  },
]
`;

exports[`emphasis code should match the consecutive occurrences 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "\`first code\` \`second code\` \`third code\`",
    "tokens": Array [
      Object {
        "raw": "\`first code\`",
        "type": "code",
        "value": "first code",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "\`second code\`",
        "type": "code",
        "value": "second code",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "\`third code\`",
        "type": "code",
        "value": "third code",
      },
    ],
    "type": "paragraph",
    "value": "\`first code\` \`second code\` \`third code\`",
  },
]
`;

exports[`emphasis code should not be deep tokenized 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "\`code with ~ gem and **bold** and *italics* and [link](https://kiranparajuli.com.np)\`",
    "tokens": Array [
      Object {
        "raw": "\`code with ~ gem and **bold** and *italics* and [link](https://kiranparajuli.com.np)\`",
        "type": "code",
        "value": "code with ~ gem and **bold** and *italics* and [link](https://kiranparajuli.com.np)",
      },
    ],
    "type": "paragraph",
    "value": "\`code with ~ gem and **bold** and *italics* and [link](https://kiranparajuli.com.np)\`",
  },
]
`;

exports[`emphasis italics should be deep tokenized 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "*italics with \`code\` and ~~strikes~~ and [link](https://kiranparajuli.com.np)*",
    "tokens": Array [
      Object {
        "raw": "*italics with \`code\` and ~~strikes~~ and [link](https://kiranparajuli.com.np)*",
        "tokens": Array [
          Object {
            "raw": "italics with ",
            "type": "text",
            "value": "italics with ",
          },
          Object {
            "raw": "\`code\`",
            "type": "code",
            "value": "code",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "~~strikes~~",
            "tokens": Array [
              Object {
                "raw": "strikes",
                "type": "text",
                "value": "strikes",
              },
            ],
            "type": "strike-through",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "[link](https://kiranparajuli.com.np)",
            "tokens": Object {
              "href": "https://kiranparajuli.com.np",
              "title": Object {
                "raw": "link",
                "tokens": Array [
                  Object {
                    "raw": "link",
                    "type": "text",
                    "value": "link",
                  },
                ],
              },
              "tooltip": undefined,
            },
            "type": "link",
          },
        ],
        "type": "italic",
      },
    ],
    "type": "paragraph",
    "value": "*italics with \`code\` and ~~strikes~~ and [link](https://kiranparajuli.com.np)*",
  },
]
`;

exports[`emphasis italics should match the consecutive occurrences 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "*first italics* *second italics* *third italics*",
    "tokens": Array [
      Object {
        "raw": "*first italics*",
        "tokens": Array [
          Object {
            "raw": "first italics",
            "type": "text",
            "value": "first italics",
          },
        ],
        "type": "italic",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "*second italics*",
        "tokens": Array [
          Object {
            "raw": "second italics",
            "type": "text",
            "value": "second italics",
          },
        ],
        "type": "italic",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "*third italics*",
        "tokens": Array [
          Object {
            "raw": "third italics",
            "type": "text",
            "value": "third italics",
          },
        ],
        "type": "italic",
      },
    ],
    "type": "paragraph",
    "value": "*first italics* *second italics* *third italics*",
  },
]
`;

exports[`emphasis link should deep tokenized the link title 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "[*italic* with **bold** and \`code\`](https://kiranparajuli.com.np)",
    "tokens": Array [
      Object {
        "raw": "[*italic* with **bold** and \`code\`](https://kiranparajuli.com.np)",
        "tokens": Object {
          "href": "https://kiranparajuli.com.np",
          "title": Object {
            "raw": "*italic* with **bold** and \`code\`",
            "tokens": Array [
              Object {
                "raw": "*italic*",
                "tokens": Array [
                  Object {
                    "raw": "italic",
                    "type": "text",
                    "value": "italic",
                  },
                ],
                "type": "italic",
              },
              Object {
                "raw": " with ",
                "type": "text",
                "value": " with ",
              },
              Object {
                "raw": "**bold**",
                "tokens": Array [
                  Object {
                    "raw": "bold",
                    "type": "text",
                    "value": "bold",
                  },
                ],
                "type": "bold",
              },
              Object {
                "raw": " and ",
                "type": "text",
                "value": " and ",
              },
              Object {
                "raw": "\`code\`",
                "type": "code",
                "value": "code",
              },
            ],
          },
          "tooltip": undefined,
        },
        "type": "link",
      },
    ],
    "type": "paragraph",
    "value": "[*italic* with **bold** and \`code\`](https://kiranparajuli.com.np)",
  },
]
`;

exports[`emphasis link should match the consecutive occurrences 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "[link](https://kiranparajuli.com.np) [link](https://kiranparajuli.com.np)",
    "tokens": Array [
      Object {
        "raw": "[link](https://kiranparajuli.com.np)",
        "tokens": Object {
          "href": "https://kiranparajuli.com.np",
          "title": Object {
            "raw": "link",
            "tokens": Array [
              Object {
                "raw": "link",
                "type": "text",
                "value": "link",
              },
            ],
          },
          "tooltip": undefined,
        },
        "type": "link",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "[link](https://kiranparajuli.com.np)",
        "tokens": Object {
          "href": "https://kiranparajuli.com.np",
          "title": Object {
            "raw": "link",
            "tokens": Array [
              Object {
                "raw": "link",
                "type": "text",
                "value": "link",
              },
            ],
          },
          "tooltip": undefined,
        },
        "type": "link",
      },
    ],
    "type": "paragraph",
    "value": "[link](https://kiranparajuli.com.np) [link](https://kiranparajuli.com.np)",
  },
]
`;

exports[`emphasis strikes should be deep tokenized 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "~~strikes with ~ gem and \`code\` and **bold** and *italics* and [link](https://kiranparajuli.com.np)~~",
    "tokens": Array [
      Object {
        "raw": "~~strikes with ~ gem and \`code\` and **bold** and *italics* and [link](https://kiranparajuli.com.np)~~",
        "tokens": Array [
          Object {
            "raw": "strikes with ~ gem and ",
            "type": "text",
            "value": "strikes with ~ gem and ",
          },
          Object {
            "raw": "\`code\`",
            "type": "code",
            "value": "code",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "**bold**",
            "tokens": Array [
              Object {
                "raw": "bold",
                "type": "text",
                "value": "bold",
              },
            ],
            "type": "bold",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "*italics*",
            "tokens": Array [
              Object {
                "raw": "italics",
                "type": "text",
                "value": "italics",
              },
            ],
            "type": "italic",
          },
          Object {
            "raw": " and ",
            "type": "text",
            "value": " and ",
          },
          Object {
            "raw": "[link](https://kiranparajuli.com.np)",
            "tokens": Object {
              "href": "https://kiranparajuli.com.np",
              "title": Object {
                "raw": "link",
                "tokens": Array [
                  Object {
                    "raw": "link",
                    "type": "text",
                    "value": "link",
                  },
                ],
              },
              "tooltip": undefined,
            },
            "type": "link",
          },
        ],
        "type": "strike-through",
      },
    ],
    "type": "paragraph",
    "value": "~~strikes with ~ gem and \`code\` and **bold** and *italics* and [link](https://kiranparajuli.com.np)~~",
  },
]
`;

exports[`emphasis strikes should match the consecutive occurrences 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "~~first strikes~~ ~~second strikes~~ ~~third strikes~~",
    "tokens": Array [
      Object {
        "raw": "~~first strikes~~",
        "tokens": Array [
          Object {
            "raw": "first strikes",
            "type": "text",
            "value": "first strikes",
          },
        ],
        "type": "strike-through",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "~~second strikes~~",
        "tokens": Array [
          Object {
            "raw": "second strikes",
            "type": "text",
            "value": "second strikes",
          },
        ],
        "type": "strike-through",
      },
      Object {
        "raw": " ",
        "type": "text",
        "value": " ",
      },
      Object {
        "raw": "~~third strikes~~",
        "tokens": Array [
          Object {
            "raw": "third strikes",
            "type": "text",
            "value": "third strikes",
          },
        ],
        "type": "strike-through",
      },
    ],
    "type": "paragraph",
    "value": "~~first strikes~~ ~~second strikes~~ ~~third strikes~~",
  },
]
`;
